src/System/refine.rs

```diff
@@ -395,3 +401,134 @@ where
 
     a == b
 }
+
+fn add_extra_inputs_outputs(
+    sys1: SystemRepresentation,
+    sys2: SystemRepresentation,
+    sys_decls: &system_declarations::SystemDeclarations,
+) -> (
+    SystemRepresentation,
+    SystemRepresentation,
+    system_declarations::SystemDeclarations,
+) {
```

Comment:
This function and probably some of the helper functions should probably be moved. As i assume adding extra input outputs is relevant for other queries than refinement.

Whether representation.rs or a specific extra_actions.rs is best is up to you.